ruby/src/api.cc

```diff
@@ -234,12 +234,20 @@ namespace leatherman { namespace ruby {
         return directories;
     }
 
+    size_t api::num2size_t(VALUE v) const
+    {
+        auto size = rb_num2ull(v);
+        if (size > numeric_limits<size_t>::max()) {
+            throw runtime_error("maximum string size exceeded, requested size was " + to_string(size));
```

Comment:
If this is gonna be in a generically-named function, it probably shouldn't have "string" in the error message?